File: ASP Core/ZenithSociety/src/ZenithWebsite/Models/Event.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;
using ZenithWebsite.Models.CustomValidation;

namespace ZenithWebsite.Models
{
    public class Event
    {
        [Key]
        public int EventId { get; set; }

        [Display(Name = "From Date & Time")]
        public DateTime EventFrom { get; set; }

        [Display(Name = "To Date & Time")]
        [IsDateAfter("EventFrom")]
        [IsSameDay("EventFrom")]
        public DateTime EventTo { get; set; }

        [Display(Name = "Creator")]
        public string UserId { get; set; }

        [ForeignKey("UserId")]
        public virtual ApplicationUser ApplicationUser { get; set; }

        [Display(Name = "Activity")]
        public int ActivityId { get; set; }

        [ForeignKey("ActivityId")]
        public Activity Activity { get; set; }

        [Display(Name = "Date Created")]
        public DateTime CreationDate { get; set; }

        [Display(Name = "Is Active")]
        public Boolean IsActive { get; set; }
    }
}

File: ASP Core/ZenithSociety/ZenithClientSite/src/app/event.ts
export class Event {
	eventId: number;
	eventFrom: Date;
	eventTo: Date;
	userId: string;
	applicationUser: string;
	activityId: number;
	activity: string;
	creationDate: Date;
	isActive: boolean

	 constructor(obj?: any ){
	 	this.eventId = obj && obj.eventId || null;
	 	this.eventFrom = obj && obj.eventFrom || null;
	 	this.eventTo = obj && obj.eventTo || null;
	 	this.userId = obj && obj.userId || null;
	 	this.applicationUser = obj && obj.applicationUser || null;
	 	this.activityId = obj && obj.activityId || null;
	 	this.activity = obj && obj.activity || null;
	 	this.creationDate = obj && obj.creationDate || null;
	 	this.isActive = obj && obj.isActive || null;
	 } 
}
